### ISSUE TITLE
Collection.Remove() Error

### ISSUE THREAD

**Issue by jarremw** [dan-eyles/sculejs#34]
I am getting the following error when trying to remove a document from a collection:

 'undefined' is not a function (evaluating 'node.detach()

I am using the Titanium storage method. I have done some debugging and have found that the node object is returning from the this.table.remove(key) as a boolean type. When I set a breakpoint the value is equal to 1. I know that my query is working because the key variable is getting populated with the right key. The error occurs at line 5859 at node.detach().

My code is similar to:

``` javascript
var collection = scule.factoryCollection('scule+titanium://test');
collection.remove({name: 'testName'},{},function(results){
    alert(results)
});
```

The code never executes to the alert(results).
